src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProvider/GetProviderQueryHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.RoatpCourseManagement.InnerApi.Requests;
+using SFA.DAS.RoatpCourseManagement.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Infrastructure;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.RoatpCourseManagement.Application.Standards.Queries.GetProvider
+{
+    public class GetProviderQueryHandler : IRequestHandler<GetProviderQuery, GetProviderResult>
+    {
+        private readonly IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _courseManagementApiClient;
+        private readonly ILogger<GetProviderQueryHandler> _logger;
+
+        public GetProviderQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> courseManagementApiClient, ILogger<GetProviderQueryHandler> logger)
+        {
+            _courseManagementApiClient = courseManagementApiClient;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderResult> Handle(GetProviderQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider request received for ukprn {ukprn}", request.Ukprn);
+
+           
+            var response = await _courseManagementApiClient.GetWithResponseCode<GetProviderResponse>(new GetProviderRequest(request.Ukprn));
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                var errorMessage =
+                   $"Response status code does not indicate success: {(int)response.StatusCode} - Provider details not found for ukprn: {request.Ukprn}";
+                _logger.LogError(errorMessage);
+                throw new HttpRequestContentException(errorMessage, response.StatusCode, response.ErrorContent);
```

Comment:
Are we allowed to throw this type of exception? MS recommends that we only use a certain set of exceptions and avoid using the rest as the framework has a meaning to them.

Reply:
I've changed to InvalidOperationException here, and will raised a tech debt to do the same elsewhere in coursemanagement outer